Subject: [PATCH] Avoid NullPointerException when updating preference intents

Comparisons changed on the intent action. Sometimes an account
preference intent does not have an action specified.

Fixes: 67075850
Test: manual - go to Settings > Users & accounts and tap an account.

diff --git a/src/com/android/settings/accounts/AccountTypePreferenceLoader.java b/src/com/android/settings/accounts/AccountTypePreferenceLoader.java
index 87aec14597..0e19dab6c0 100644
--- a/src/com/android/settings/accounts/AccountTypePreferenceLoader.java
+++ b/src/com/android/settings/accounts/AccountTypePreferenceLoader.java
@@ -34,6 +34,7 @@
 import android.support.v7.preference.Preference.OnPreferenceClickListener;
 import android.support.v7.preference.PreferenceGroup;
 import android.support.v7.preference.PreferenceScreen;
+import android.text.TextUtils;
 import android.util.Log;
 
 import com.android.settings.R;
@@ -135,8 +136,8 @@ public void updatePreferenceIntents(PreferenceGroup prefs, final String acccount
                 // startPreferencePanel() there. In order to inject the title string there, more
                 // dirty further hack is still needed. It's much easier and cleaner to listen to
                 // preference click event here directly.
-                if (intent.getAction().equals(
-                    android.provider.Settings.ACTION_LOCATION_SOURCE_SETTINGS)) {
+                if (TextUtils.equals(intent.getAction(),
+                        android.provider.Settings.ACTION_LOCATION_SOURCE_SETTINGS)) {
                     // The OnPreferenceClickListener overrides the click event completely. No intent
                     // will get fired.
                     pref.setOnPreferenceClickListener(new FragmentStarter(
